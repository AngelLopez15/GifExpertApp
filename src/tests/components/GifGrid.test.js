import React from 'react';
import '@testing-library/jest-dom'

// traemos el componente por que vamos a simular la peticion de imagenes
import {useFetchGifs} from '../../hooks/useFetchGifs'
// para simular la llamada al archivo
jest.mock('../../hooks/useFetchGifs')

const { shallow } = require("enzyme")
const { GifGrid } = require("../../components/GifGrid")

describe('Pruebas en el <GifGrid />', () => {
    
    //Las pruebas unitarias significa que debemos comprobar los componenetes de 
    // manera aislada

    const category = 'One punch'

    test('debe de mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        })

        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
    })
    
    test('debe de mostrar items cuando se cargan las imagenes que trae el useFetchGifs', () => {
        
        const gifs = [{
            id:'ABC',
            url:'https://peticion/aalgunlado/cosa.gif',
            title:'titulo'
        }]

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        })

        const wrapper = shallow(<GifGrid category={category} />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
    

})
