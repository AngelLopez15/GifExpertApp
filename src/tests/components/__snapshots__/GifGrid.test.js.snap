// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> debe de mostrar items cuando se cargan las imagenes que trae el useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__backInLeft"
  >
    One punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="titulo"
      url="https://peticion/aalgunlado/cosa.gif"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__backInLeft"
  >
    One punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
